File: Arise.FileSyncer.Service/Ipc/Messages/SendProfileMessage.cs
using System;

namespace Arise.FileSyncer.Service.Ipc.Messages
{
    public class SendProfileMessage : IpcMessage
    {
        public override string Command => "sendProfile";

        public Guid ConnectionId { get; set; }
        public Guid ProfileId { get; set; }

        internal override void Process(IpcController ipc)
        {
            ipc.Service.Peer.ShareProfile(ConnectionId, ProfileId);
        }
    }
}

File: Arise.FileSyncer.TauriUI/src/ipc/messages/sendProfileMessage.ts
import { IpcMessage } from "../ipcMessage"

export const Command = "sendProfile"

export interface Message extends IpcMessage, Data { }

export function Create(data: Data): Message {
    const message: Message = {
        Command: Command,
        ConnectionId: data.ConnectionId,
        ProfileId: data.ProfileId
    }

    return message
}

export interface Data {
    ConnectionId: string,
    ProfileId: string,
}
